packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx

```diff
@@ -271,6 +273,17 @@ export const MultiSectionDigitalClock = React.forwardRef(function MultiSectionDi
 
           return !containsValidTime({ start, end }) || !isValidValue(rawValue);
         }
+        case 'meridiem': {
+          const start = adapter.setSeconds(
+            adapter.setMinutes(
+              adapter.setHours(adapter.startOfDay(valueOrReferenceDate), rawValue),
+              0,
+            ),
+            0,
+          );
+          const end = adapter.addSeconds(adapter.addMinutes(adapter.addHours(start, 11), 59), 59);
+          return !containsValidTime({ start, end });
```

Comment:
We should include the check for valid values here as well for safety reasons.

```suggestion
          return !containsValidTime({ start, end }) || !isValidValue(rawValue);
```